alert/expiring.py

```diff
@@ -0,0 +1,80 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Search for regression in a histogram dump directory produced by the
+# node exporter.
+
+import json
+import urllib2
+from datetime import datetime, timedelta
+
+from bs4 import BeautifulSoup
+from mail import send_ses
+
+HISTOGRAMS_FILE = "Histograms.json"
+EMAIL_TIME_BEFORE = timedelta(weeks=1)
+FROM_ADDR = "cerberus-alert@mozilla.com"
+
+def get_future_release_dates():
+    # scrape for future release date tables
+    response = json.loads(urllib2.urlopen("https://wiki.mozilla.org/api.php?action=parse&format=json&page=RapidRelease/Calendar").read())
+    soup = BeautifulSoup(response["parse"]["text"]["*"])
+    table = soup.find(id="Future_branch_dates").find_parent("h2").find_next_sibling("table")
+    result = {}
+    for i, row in enumerate(table.find_all("tr")):
+        if i == 0: continue # skip the header row
+        version = list(row.find_all("td"))[-1].string.strip().replace("Firefox ", "")
+        version = version.split(".")[0] # only keep the major version
+        date_string = list(row.find_all("th"))[-1].string.strip(" \t\r\n*")
+        try: result[version] = datetime.strptime(date_string, "%Y-%m-%d")
+        except ValueError: pass
+    return result
+
+def is_expiring(histogram_entry, now, release_dates):
+    expiry = histogram_entry.get("expires_in_version", "never").split(".")[0]
+
+    # check if the version is unknown or far into the future
+    if expiry in {"never", "default"}: return False
+    try:
+        max_release = max(map(int, release_dates.keys()))
+        if int(expiry) > max_release: return False
+    except ValueError: pass
+    if expiry not in release_dates: return True
+    
+    # known version, check if it is close to release
+    return release_dates[expiry] - now < EMAIL_TIME_BEFORE
```

Comment:
This takes in consideration only the major version; the version comparison needs to be compatible with our versioning scheme, i.e. versions such as "40.0a1" (v40 nightly) have to be properly supported.